google/cloud/forseti/services/model/importer/importer.py

```diff
@@ -1343,16 +1278,16 @@ def _is_role_unique(self, role_name):
             return False
         return True
 
-    def _add_to_cache(self, resource):
+    def _add_to_cache(self, resource, resource_id):
         """Add a resource to the cache for parent lookup.
 
         Args:
             resource (object): Resource to put in the cache.
+            resource_id (int): The database key for the resource.
         """
 
-        type_name = resource.type_name
         full_res_name = resource.full_name
-        self.resource_cache[type_name] = (resource, full_res_name)
+        self.resource_cache[resource_id] = (resource, full_res_name)
```

Comment:
qq: can you please explain a bit why this change from `type_name` to `resource_id` is made?

Reply:
This change is made because we now track the relationship between parent and child resources using resource.parent_id = resource.id. The is no longer a parent_resource_type and parent_resource_id to join on.